src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/validation/actions/OnReceivedYFlowResourcesAction.java

```diff
@@ -40,24 +42,27 @@ public class OnReceivedYFlowResourcesAction extends
     public void perform(State from, State to, Event event, YFlowValidationContext context,
                         YFlowValidationFsm stateMachine) {
         MessageData data = context.getSpeakerResponse();
-        if (data instanceof SwitchFlowEntries) {
-            SwitchFlowEntries switchFlowEntries = (SwitchFlowEntries) data;
-            log.info("Switch rules received for switch {}", switchFlowEntries.getSwitchId());
-            stateMachine.getReceivedRules().add(switchFlowEntries);
+        if (data instanceof FlowDumpResponse) {
+            FlowDumpResponse flowDumpResponse = (FlowDumpResponse) data;
+            log.info("Switch rules received for switch {}", Optional.ofNullable(flowDumpResponse.getSwitchId())
+                    .map(SwitchId::toString).orElse("No switch Id"));
```

Comment:
Optional is unnecessary here. You can just use `flowDumpResponse.getSwitchId().`
`log.info("Switch rules received for switch {}", flowDumpResponse.getSwitchId()` will just call SwitchId.toString or print `null`

Reply:
fixed